kernel/src/main/scala/cats/kernel/std/array.scala

```diff
@@ -0,0 +1,73 @@
+package cats.kernel
+package std
+
+import scala.{ specialized => sp }
+
+package object array extends ArrayInstances
```

Comment:
Since cats focuses on _pure_ functional programming, would it make sense for `Array` instances to not be part of cats-kernel and instead stay in algebra or something?